src/App.tsx

```diff
@@ -1,61 +1,164 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+interface Todo {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+}
+
+interface User {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+}
+
+function getUserIDByName(arr: User[], name: string) {
+  const user = arr.find(item => item.name === name);
+
+  if (user !== undefined) {
+    return user.id;
+  }
+
+  return 1;
+}
+
+function getRandomId(): number {
+  return Math.floor(Math.random() * 100000);
+}
 
 export const App = () => {
+  const users: User[] = usersFromServer;
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+
+  const [title, setTitle] = useState('');
+  const [name, setName] = useState('0');
+
+  const newTodo = {
+    id: getRandomId(),
+    title,
+    completed: false,
+    userId: getUserIDByName(users, name),
+  };
+
+  const [titleError, setTitleError] = useState(false);
+  const [nameError, setNameError] = useState(false);
+
+  const handleTitleError = () => {
+    if (!title) {
+      setTitleError(true);
+    } else {
+      setTitleError(false);
+    }
+  };
+
+  const handleNameError = () => {
+    if (name === '0') {
+      setNameError(true);
+    } else {
+      setNameError(false);
+    }
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
```

Comment:
```suggestion
  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
    setTitleError(false)
  };
```
if the user enters something, let's hide the error